traj/src/traj/cubic_eq_roots.py

```diff
@@ -0,0 +1,197 @@
+#!/usr/bin/env python
+
+import math
+
+#to find the real root/roots of a cubic equation
+def real_roots_cubic_eq ( a,  b,  c,  d):
+#    print "real_roots_cubic_eq: a, b, c, d= "
+#    print  a,  b, c, d
+    
+    rt1=0
+    rt2=0
+    rt3=0
+    if (a == 0.0000000):
+#        print "The coefficient of the cube of x is 0. Please use the utility for a SECOND degree quadratic. No further action taken."
+        rt1, rt2, n_rts = quad_eq_real_root ( b, c, d)
+        return rt1, rt2, -100, n_rts
+    
+    if ( abs(d) < 1e-10):
+#        print "One root is 0. Now divide through by x and use the utility for a SECOND degree quadratic to solve the resulting equation for the other two roots. No further action taken."
+        rt1 = 0
+        rt2, rt3, n_rts = quad_eq_real_root (a, b, c)
+        return rt1, rt2, rt3, n_rts+1
+    
+    b /= a
+    c /= a
+    d /= a
+#    double disc, q, r, dum1, s, t, term1, r13;
+    q = (3.0*c - (b*b))/9.0
+    r = -(27.0*d) + b*(9.0*c - 2.0*(b*b))
```

Comment:
Nitpick: would be good to run a formatter on the code to standardize the spacing.